src/Menu/HomePageRenderer.cpp

```diff
@@ -0,0 +1,449 @@
+	}
+
+	// Center all content
+	float windowWidth = ImGui::GetWindowWidth();
+
+	// Welcome title - centered
+	const char* welcomeTitle = "Welcome to Community Shaders!";
+	float welcomeTitleWidth = ImGui::CalcTextSize(welcomeTitle).x;
+	ImGui::SetCursorPosX((windowWidth - welcomeTitleWidth) * 0.5f);
+	ImGui::Text("%s", welcomeTitle);
+
+	// Version text - centered
+	const char* versionText = "This appears to be a new install, update, or reinstallation of Community Shaders.";
+	float versionWidth = ImGui::CalcTextSize(versionText).x;
+	ImGui::SetCursorPosX((windowWidth - versionWidth) * 0.5f);
+	ImGui::Text("%s", versionText);
+
+	ImGui::Spacing();
+
+	// Description - centered
+	const char* description = "Please select a hotkey to access the menu:";
+	float descWidth = ImGui::CalcTextSize(description).x;
+	ImGui::SetCursorPosX((windowWidth - descWidth) * 0.5f);
+	ImGui::Text("%s", description);
+
+	// Hotkey selection - centered
+	static int selectedKey = 0;
+	const char* keyOptions[] = {
+		"END (default)", "INSERT", "HOME", "DELETE",
+		"PAGE UP", "PAGE DOWN", "F9", "F10", "F11", "F12"
+	};
+
+	const uint32_t keyValues[] = {
+		VK_END, VK_INSERT, VK_HOME, VK_DELETE,
+		VK_PRIOR, VK_NEXT, VK_F9, VK_F10, VK_F11, VK_F12
+	};
+
+	// Center the dropdown
+	float comboWidth = 160.0f;
+	ImGui::SetCursorPosX((windowWidth - comboWidth) * 0.5f);
+	ImGui::SetNextItemWidth(comboWidth);
+	ImGui::Combo("##HotkeyDropdown", &selectedKey, keyOptions, IM_ARRAYSIZE(keyOptions));
+
+	ImGui::Spacing();
+
+	// "You can change this later" text - centered
+	const char* laterText = "You can change this later in General > Keybindings.";
+	float laterWidth = ImGui::CalcTextSize(laterText).x;
+	ImGui::SetCursorPosX((windowWidth - laterWidth) * 0.5f);
+	ImGui::Text("%s", laterText);
+
+	ImGui::Spacing();
+
+	// Center the continue button
+	float buttonWidth = 120.0f;
+	ImGui::SetCursorPosX((windowWidth - buttonWidth) * 0.5f);
+
+	// Check for Enter or Escape key first
+	bool shouldClose = ImGui::IsKeyPressed(ImGuiKey_Enter) || ImGui::IsKeyPressed(ImGuiKey_Escape);
+
+	if (ImGui::Button("Continue", ImVec2(buttonWidth, 30)) || shouldClose) {
+		// Apply the selected hotkey
+		if (menu) {
+			menu->GetSettings().ToggleKey = keyValues[selectedKey];
+		}
+		MarkFirstTimeSetupComplete();
+	}
+
+	// Center the help text
+	const char* helpText = "(Press Enter or Escape to continue)";
+	float helpWidth = ImGui::CalcTextSize(helpText).x;
+	ImGui::SetCursorPosX((windowWidth - helpWidth) * 0.5f);
+	ImGui::TextDisabled("%s", helpText);
+
+	ImGui::PopStyleVar(2);  // Pop WindowRounding and WindowBorderSize
+	ImGui::End();
+}
+
+bool HomePageRenderer::ShouldShowFirstTimeSetup()
+{
+	// Check if already completed this session
+	if (isFirstTimeSetupShown) {
+		return false;
+	}
+
+	// Check if first-time setup has been completed by looking at UserSettings.json
+	std::filesystem::path userSettingsPath = "Data\\SKSE\\Plugins\\CommunityShaders\\UserSettings.json";
```

Comment:
We have helper functions to find paths, etc. Don't introduce yet another hard coded path.